app/src/main/java/com/willowtree/vocable/settings/EditPhrasesFragment.kt

```diff
@@ -93,6 +95,42 @@ class EditPhrasesFragment: BaseFragment() {
         return binding?.root
     }
 
+    private fun showDeletePhraseDialog(phraseView: PhraseEditLayoutBinding) {
+        setSettingsButtonsEnabled(false)
+        binding?.deleteConfirmation?.dialogTitle?.text = getString(R.string.are_you_sure)
+        binding?.deleteConfirmation?.dialogMessage?.text = getString(R.string.delete_warning)
+        binding?.deleteConfirmation?.dialogPositiveButton?.let {
+            it.text = getString(R.string.delete)
+            it.action = {
+                editPhrasesViewModel.deletePhrase(phraseView.phraseEditText.tag as Phrase)
+                toggleDialogVisibility(false)
+                setSettingsButtonsEnabled(true)
+            }
+        }
+        binding?.deleteConfirmation?.dialogNegativeButton?.let {
+            it.text = getString(R.string.settings_dialog_cancel)
+            it.action = {
+                toggleDialogVisibility(false)
+                setSettingsButtonsEnabled(true)
+            }
+        }
+        toggleDialogVisibility(true)
+    }
+
+    private fun setSettingsButtonsEnabled(enable: Boolean) {
+        editPhrasesViewModel.setEditButtonsEnabled(enable)
+
+    }
+
+    private fun toggleDialogVisibility(visible: Boolean) {
+        binding?.deleteConfirmation?.root?.let {
+            it.isVisible = visible
+            it.post {
```

Comment:
You don't need the `post` here so you should be able to get rid of the `let` too